src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -281,32 +256,77 @@ function joinChunks(A: Chunk, B: Chunk): Chunk {
     entities: A.entities.concat(B.entities),
     blocks: A.blocks.concat(B.blocks),
   };
-}
+};
 
 /**
  * Check to see if we have anything like <p> <blockquote> <h1>... to create
  * block tags from. If we do, we can use those and ignore <div> tags. If we
  * don't, we can treat <div> tags as meaningful (unstyled) blocks.
  */
-function containsSemanticBlockMarkup(
+const containsSemanticBlockMarkup = (
   html: string,
   blockTags: Array<string>,
-): boolean {
+): boolean => {
   return blockTags.some(tag => html.indexOf('<' + tag) !== -1);
-}
+};
 
-function hasValidLinkText(link: Node): boolean {
+const hasValidLinkText = (link: Node): boolean => {
   invariant(
     link instanceof HTMLAnchorElement,
     'Link must be an HTMLAnchorElement.',
   );
-  var protocol = link.protocol;
+  const protocol = link.protocol;
   return (
     protocol === 'http:' || protocol === 'https:' || protocol === 'mailto:'
   );
-}
+};
+
+const getWhitespaceChunk = (inEntity: ?string): Chunk => {
+  return {
+    ...EMPTY_CHUNK,
+    text: SPACE,
+    inlines: [OrderedSet()],
+    entities: new Array(1).map(none => (inEntity ? inEntity : none)),
+  };
+};
+
+const getSoftNewlineChunk = (): Chunk => {
+  return {
+    ...EMPTY_CHUNK,
+    text: '\n',
+    inlines: [OrderedSet()],
+    entities: new Array(1),
+  };
+};
+
+const getChunkedBlock = (props: Object = {}): Block => {
+  return {
+    ...EMPTY_BLOCK,
+    ...props,
+  };
+};
 
-function genFragment(
+const getBlockDividerChunk = (
+  block: DraftBlockType,
+  depth: number,
+  parentKey: ?string = null,
+): Chunk => {
+  return {
+    text: '\r',
+    inlines: [OrderedSet()],
+    entities: new Array(1),
+    blocks: [
+      getChunkedBlock({
+        parent: parentKey,
+        key: generateRandomKey(),
```

Comment:
The change in here is that when we create a `block chunk` we will always want a key since we will use that latter to build the tree data blocks.